server/README.md

```diff
@@ -0,0 +1,472 @@
+- A block is identified by a client-provided UUID, then there is no risk for double-creation
+  under concurrency (a trivial uniqueness check on insertion prevents this).
+- Block creation date must be more recent than the last certificate in `common` topic and
+  in the `realm` topic corresponding to the block's realm.
+
+> Notes:
+>
+> - It could be tempting to only lock the `realm` topic (given it's this topic that contains
+>   the certificate describing the role of a user in the realm). However a revoked user is also
+>   forbidden from adding data, hence making it mandatory to also lock the `common` topic here.
+> - Uploading blocks (and vlobs) are the most common task on the server, so we may want to
+>   optimise it in the future to avoid having to lock two topics.
+>   A possible solution would be to have an optimistic locking approach: first read the
+>   last timestamp in the topics, then do the checks, then on the final data insertion check
+>   again the timestamp haven't changed (and if they have we re-do everything from the start).
+
+### Vlobs
+
+Vlobs are encrypted metadata that belong to a given realm. Unlike block they can be
+modified and hence have a version field that must increase linearly.
+
+Concurrency & causality is handle in a similar fashion as for blocks.
+
+TODO: explain checkpoint logic, and concurrent vlob insertion causing unique violation on checkpoint
+
+> Note:
+>
+> This may seem counter-intuitive given vlobs require a database write, but blocks require
+> both a database write and upload to the object storage. However you can think of both vlob
+> and block upload as:
+>
+> 1) Fetch info from database.
+> 2) Do stuff in Python.
+> 3) Store the vlob/block in the database.
+>
+> In the case of vlobs, step 2 is only about checking access rights and vlob existence.
+> On the other hand blocks also includes the write to the object storage.
+>
+> However this is irrelevant from the database point of view: in both case step 3 is only
+> allowed if data in step 1 hasn't changed.
+>
+> Obviously this approach means we can end up with data in the  object storage while the
+> SQL transaction got cancelled, however this is acceptable:
+>
+> - Given the block ID is determined client-side, re-trying to upload the block will lead
+>   to another write in the object storage at the same location.
+> - The object storage follows an eventual consistency strategy, hence the new block upload
+>   is guaranteed to overwrite the previous one.
+> - The only issue would be if multiple attempts of the same block ID would be done with
+>   different data on each try. In this case eventual consistency means it is possible
+>   for a short period of time to get a previous value when querying the block.
+>   However this is purely theoretical since block ID is random (hence there should be
+>   no concurrency between clients) and a client considers a block as immutable.
+
+### Invitations
```

Comment:
This section should be updated once https://github.com/Scille/parsec-cloud/issues/9302 is treated.